### ISSUE TITLE
data model: howto use MCC and/or MNC?

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe-engine#337]
Assuming we decide that https://github.com/ooni/probe/issues/1014 looks good, that we understand the implications of roaming https://github.com/ooni/probe/issues/1015, that we discuss this with the community and the discussion looks good https://github.com/ooni/ooni.org/issues/389, then we need to understand how to change the data model to include MCC and/or MNC.

**Comment by fortuna** [comment]
At least initially, I'd suggest saving all of them:
- IP location
- SIM operator location
- Network location

This way the data consumer can choose how to use them.

To figure out how to use them we need to think about the semantics of what we want to measure.
What are the factors that are relevant to censorship? 

If an Iranian or Chinese is roaming in the US, do they experience censorship?
If a Italian is roaming in Iran or China, do they experience censorship?
We may not actually know, and we want to figure that out. That's why having all the locations help.

I have the impression that the defining factor is the SIM operator location most of the time, based on stories I've heard. We may want to use that as the "best guess" to assign censorship to locations.

One thing to keep in mind is that the user may be connected to cell for telephony, and to wifi for data. In that case the network location would be better than the sim location, since the network location is a good proxy for the user's physical location, which would match the wifi location.

So in short:
- Use mobile location
  - If the user is on Wifi, use the network location
  - else, use the SIM operator location
- If mobile location not available, use the IP location.


Another fun thing to explore: evaluate how good the IP location is by comparing to the mobile location.

**Comment by fortuna** [comment]
I'd be happy to participate in any meeting where you discuss this.

**Comment by fortuna** [comment]
FYI, I added a discussion on this topic to the design doc I'm working on, including a proposal for how to use them: https://docs.google.com/document/d/1jcidvZGxBlucyLivAtvwrCidkIgb3yV6bhQIH_jf21M/edit#heading=h.ohm2kl35kyig
